Parsec 6.0 breaks the build

compiling the repo with the latest parsec version cause a compile error:
```bash
make  all-recursive
make[1]: Entering directory '/workspace/template-nix/vdi-stream-client-0.2.0'
Making all in src
make[2]: Entering directory '/workspace/template-nix/vdi-stream-client-0.2.0/src'
gcc -DHAVE_CONFIG_H -I. -I..    -I/usr/include/libusb-1.0 -I/usr/include/libusb-1.0    -D_REENTRANT -I/usr/include/SDL2 -D_REENTRANT -I/usr/include/SDL2 -g -O2 -c -o vdi_stream_client-parsec.o `test -f 'parsec.c' || echo './'`parsec.c
parsec.c: In function ‘vdi_stream_client__cursor’:
parsec.c:76:14: error: ‘ParsecCursor’ {aka ‘struct ParsecCursor’} has no member named ‘modeUpdate’; did you mean ‘imageUpdate’?
   76 |  if (cursor->modeUpdate == SDL_TRUE) {
      |              ^~~~~~~~~~
      |              imageUpdate
make[2]: *** [Makefile:429: vdi_stream_client-parsec.o] Error 1
make[2]: Leaving directory '/workspace/template-nix/vdi-stream-client-0.2.0/src'
make[1]: *** [Makefile:385: all-recursive] Error 1
make[1]: Leaving directory '/workspace/template-nix/vdi-stream-client-0.2.0'
make: *** [Makefile:326: all] Error 2
```

looking at their SDK changes with [previous version ( 5.0 )](https://github.com/parsec-cloud/parsec-sdk/compare/5.0...6.0#diff-6bdd288efb95c09792c5785e885b9f33b36933430d5010d9643bd2f75d870e21R679), they replace `modeUpdate` with `hidden`. changing the code will compile it fine.

--- a/README.md
+++ b/README.md
@@ -256,6 +256,11 @@ solution requires a host service running inside the Windows environment.
 Hardware encoding is supported by FFmpeg for the major GPU vendors. VDI Stream
 Host application seems reasonable.
 
+# Thanks
+
+A special thanks goes to Parsec Cloud, Inc. as they provided a Parsec Warp
+subscription to support this project.
+
 # Support
 
 If you want to help out with development without providing code yourself, you

--- a/configure.ac
+++ b/configure.ac
@@ -49,6 +49,7 @@ if test "x$enable_internal_parsec_sdk" != "xyes"; then
 	# checking for external parsec sdk.
 	AC_CHECK_HEADER([parsec/parsec.h], [], [AC_MSG_ERROR([*** parsec/parsec.h is required, install parsec-sdk header files])])
 	AC_CHECK_LIB([parsec], [ParsecInit], [], [AC_MSG_ERROR([*** ParsecInit is required, install parsec-sdk library files])])
+	AC_CHECK_MEMBER([struct ParsecCursor.hidden], [], [AC_MSG_ERROR([*** struct ParsecCursor.hidden is required, install parsec-sdk >= 6.0])], [[#include <parsec/parsec.h>]])
 fi
 
 # condition for parsec sdk to usage.

--- a/src/parsec.c
+++ b/src/parsec.c
@@ -52,7 +52,7 @@ static void vdi_stream_client__clipboard(struct parsec_context_s *parsec_context
 }
 
 /* parsec cursor event. */
-static void vdi_stream_client__cursor(struct parsec_context_s *parsec_context, ParsecCursor *cursor, Uint32 buffer_key, SDL_bool grab) {
+static void vdi_stream_client__cursor(struct parsec_context_s *parsec_context, ParsecCursor *cursor, Uint32 buffer_key, SDL_bool grab, SDL_bool grab_forced) {
 	if (cursor->imageUpdate == SDL_TRUE) {
 		Uint8 *image = ParsecGetBuffer(parsec_context->parsec, buffer_key);
 
@@ -73,27 +73,20 @@ static void vdi_stream_client__cursor(struct parsec_context_s *parsec_context, P
 		}
 	}
 
-	if (cursor->relative == SDL_TRUE || cursor->hidden == SDL_TRUE) {
-		if (parsec_context->focus == SDL_TRUE) {
-			if (SDL_GetRelativeMouseMode() == SDL_TRUE && cursor->relative == SDL_FALSE) {
-				SDL_SetRelativeMouseMode(SDL_FALSE);
-				SDL_ShowCursor(SDL_TRUE);
-				SDL_WarpMouseInWindow(parsec_context->window, cursor->positionX, cursor->positionY);
-				if (parsec_context->pressed == SDL_FALSE && grab == SDL_FALSE) {
-					SDL_SetWindowTitle(parsec_context->window, "VDI Stream Client");
-				}
-			}
-			if (SDL_GetRelativeMouseMode() == SDL_FALSE && cursor->relative == SDL_TRUE) {
-				SDL_ShowCursor(SDL_FALSE);
-				SDL_SetRelativeMouseMode(SDL_TRUE);
-				if (parsec_context->pressed == SDL_FALSE) {
-					SDL_SetWindowTitle(parsec_context->window, "VDI Stream Client (Press Ctrl+Alt to release grab)");
-				}
-			}
-			if (SDL_GetRelativeMouseMode() == SDL_FALSE && cursor->relative == SDL_TRUE && parsec_context->pressed == SDL_TRUE) {
-				SDL_ShowCursor(SDL_FALSE);
-				SDL_SetRelativeMouseMode(SDL_TRUE);
-			}
+	if (cursor->relative == SDL_TRUE && cursor->hidden == SDL_TRUE) {
+		SDL_ShowCursor(SDL_FALSE);
+		SDL_SetRelativeMouseMode(SDL_TRUE);
+		if (parsec_context->pressed == SDL_FALSE && grab_forced == SDL_FALSE) {
+			SDL_SetWindowTitle(parsec_context->window, "VDI Stream Client (Press Ctrl+Alt to release grab)");
+		}
+		parsec_context->relative = cursor->relative;
+	}
+	if (cursor->relative == SDL_FALSE && cursor->hidden == SDL_FALSE) {
+		SDL_SetRelativeMouseMode(SDL_FALSE);
+		SDL_ShowCursor(SDL_TRUE);
+		SDL_WarpMouseInWindow(parsec_context->window, cursor->positionX, cursor->positionY);
+		if (parsec_context->pressed == SDL_FALSE && grab == SDL_FALSE && grab_forced == SDL_FALSE) {
+			SDL_SetWindowTitle(parsec_context->window, "VDI Stream Client");
 		}
 		parsec_context->relative = cursor->relative;
 	}
@@ -660,7 +653,7 @@ Sint32 vdi_stream_client__event_loop(vdi_config_s *vdi_config) {
 		for (ParsecClientEvent event; ParsecClientPollEvents(parsec_context.parsec, 0, &event);) {
 			switch (event.type) {
 				case CLIENT_EVENT_CURSOR:
-					vdi_stream_client__cursor(&parsec_context, &event.cursor.cursor, event.cursor.key, vdi_config->grab);
+					vdi_stream_client__cursor(&parsec_context, &event.cursor.cursor, event.cursor.key, vdi_config->grab, grab_forced);
 					break;
 				case CLIENT_EVENT_USER_DATA:
 					if (vdi_config->clipboard == 1) {

--- a/src/parsec.c
+++ b/src/parsec.c
@@ -73,7 +73,7 @@ static void vdi_stream_client__cursor(struct parsec_context_s *parsec_context, P
 		}
 	}
 
-	if (cursor->modeUpdate == SDL_TRUE) {
+	if (cursor->relative == SDL_TRUE || cursor->hidden == SDL_TRUE) {
 		if (parsec_context->focus == SDL_TRUE) {
 			if (SDL_GetRelativeMouseMode() == SDL_TRUE && cursor->relative == SDL_FALSE) {
 				SDL_SetRelativeMouseMode(SDL_FALSE);

